### DISCUSSION TITLE
Simple Terminal

### DISCUSSION THREAD

**Discussion by camenduru** [Comfy-Org/ComfyUI#130] [Q&A] [answered]
Hi @comfyanonymous 👋 I am trying to make a simple terminal node it is working but I am getting error is it possible to run node independent 

![Screenshot 2023-03-17 124048](https://user-images.githubusercontent.com/54370274/225869089-444b86de-356a-4fe7-9424-b46322a88a6d.png)

```py
from subprocess import getoutput

class Terminal:
    def __init__(self):
        pass
    
    @classmethod
    def INPUT_TYPES(s):
        return {"required": {"image": ("IMAGE",), "text": ("STRING", {"multiline": True})}}
    RETURN_TYPES = ("IMAGE",)
    FUNCTION = "execute"
    def execute(self, image, text):
        out = getoutput(f"{text}")
        print(out)
        return (image, )

    CATEGORY = "Tools"

NODE_CLASS_MAPPINGS = {
    "Terminal": Terminal
}
```

```py
got prompt
Name: torch
Version: 1.13.1+cu116
Summary: Tensors and Dynamic neural networks in Python with strong GPU acceleration
Home-page: https://pytorch.org/
Author: PyTorch Team
Author-email: [packages@pytorch.org](mailto:packages@pytorch.org)
License: BSD-3
Location: /usr/local/lib/python3.9/dist-packages
Requires: typing-extensions
Required-by: accelerate, fastai, open-clip-torch, pytorch-lightning, timm, torchaudio, torchdiffeq, torchmetrics, torchsde, torchtext, torchvision, triton, xformers
Traceback (most recent call last):
  File "/content/ComfyUI/execution.py", line 174, in execute
    executed += recursive_execute(self.server, prompt, self.outputs, x, extra_data)
  File "/content/ComfyUI/execution.py", line 54, in recursive_execute
    executed += recursive_execute(server, prompt, outputs, input_unique_id, extra_data)
  File "/content/ComfyUI/execution.py", line 64, in recursive_execute
    if "ui" in outputs[unique_id] and server.client_id is not None:
  File "/usr/local/lib/python3.9/dist-packages/torch/_tensor.py", line 985, in __contains__
    raise RuntimeError(
RuntimeError: Tensor.__contains__ only supports Tensor or scalar, but you passed in a <class 'str'>.
```

**Comment by m957ymj75urz** [comment, marked as answer]
`return (image, )` maybe ?

**Reply by camenduru** [reply]
wooohooo I am not getting any errors 🥳 thanks @m957ymj75urz ❤

**Comment by WASasquatch** [comment]
We need to be able to output text and other data back to the node. Having to output to console is such an annoyance. I've accidently closed out Comfy twice now bringing it up and minimizing constantly

**Reply by comfyanonymous** [reply]
If you do this: https://github.com/comfyanonymous/ComfyUI/blob/master/nodes.py#L772

It will send the data back to the UI. Note that it only works for OUTPUT_NODE=True nodes right now.

**Reply by WASasquatch** [reply]
> If you do this: https://github.com/comfyanonymous/ComfyUI/blob/master/nodes.py#L772
> 
> It will send the data back to the UI. Note that it only works for OUTPUT_NODE=True nodes right now.

I assume `OUTPUT_NODE=True` is basically a flag for output to GUI?

**Reply by comfyanonymous** [reply]
No, it's actually used to set the final nodes in the graph that ComfyUI will actually try to execute.

When you queue a prompt ComfyUI will try to execute each output node one by one and by doing that will execute all the parents of these nodes.

**Reply by WASasquatch** [reply]
> No, it's actually used to set the final nodes in the graph that ComfyUI will actually try to execute.
> 
> When you queue a prompt ComfyUI will try to execute each output node one by one and by doing that will execute all the parents of these nodes.

Probably going to immediately abuse it and break something, but going to give it a try. lol

**Reply by WASasquatch** [reply]
Hmm, I couldn't get it to show any text. Is it only hard set for showing images? 

I tried adding `{"ui":{"prompt":new_text}}`

**Reply by comfyanonymous** [reply]
yes you'll have to write some code in the frontend to handle things other than images.

**Reply by WASasquatch** [reply]
> yes you'll have to write some code in the frontend to handle things other than images.

Do you know where I should look? I tried briefly searching for image output but had a lot to look through. I may be able to PR that support for future internal nodes and custom nodes.
